### ISSUE TITLE
Properties are not checked properly

### ISSUE THREAD

**Issue by Shutnik** [rossPatton/stylint#231]
➜ cat test.styl 
```
::-webkit-scrollbar
  color:red

a::-webkit-scrollbar
  color:green

// but

::selection
  color:blue

a::selection
  color:yellow
```
➜ stylint test.styl 
```
Warning: property is not valid
File: test.styl
Line: 1: ::-webkit-scrollbar


Stylint: 0 Errors.
Stylint: 1 Warnings.
```

**Comment by rossPatton** [comment]
this value is probably just not on the whitelist, should be easy to update

**Comment by Shutnik** [comment]
nope, this value is [whitelisted](https://github.com/rossPatton/stylint/blob/master/src/data/valid.json#L716)

**Comment by thybzi** [comment]
For me, the following selectors throw `Warning: property is not valid`
```
Warning: property is not valid
File: resources/styl/_reset.styl
Line: 45: :focus

Warning: property is not valid
File: resources/styl/_reset.styl
Line: 98: meta
```

Both represent rule selectors with (bad and ugly) styles below:
```stylus
:focus
  outline: 0

meta
  display: none !important // @stylint ignore
```

**Comment by nayaabkhan** [comment]
@rossPatton I think I've figured what's going on here.

Psuedo selectors validation is totally not working:

```js
return arr[0] === html || this.checkPseudo( arr[0], html, validJSON )
```

For matching Pseudo selectors, we simply try to match with the HTML tag in the selector itself. So, even `a::-anything-here` validates too! And hence `a::-webkit-scrollbar` and `a::selection` validate.

The reason why `::-webkit-scrollbar` doesn't validate is that we use this check in `checkPseudo`:

```js
return prop === html + pseudo
```

So we essentially make it mandatory for a pseudo to have an element prefixed. But even if we remove the concatenation of `html` in the comparison it wouldn't work. We strip off `:` from our tokens before validating a line:

```js
var arr = this.splitAndStrip( new RegExp( /[\s\t,:]/ ), line ) // 1
```

And so our pseudo selectors get stripped off their `::`.

And the reason why `::selection` validates is that it is listed in the `css` list as `selection` without any colons. The `::selection` or `:selection` entries in `valid.json` play no role in validating it whatsoever.

I am trying to look into a solution for this. Let me know if you have anything to suggest.

**Comment by nayaabkhan** [comment]
@thybzi It is interesting you want to style the `meta` tags? Never thought it is possible to style them. Can you share in what scenario would you want to style them?

**Comment by thybzi** [comment]
@nayaabkhan 
> It is interesting you want to style the meta tags? Never thought it is possible to style them. Can you share in what scenario would you want to style them?

I had the following case (simplified):
```html
<div class="container">
   <span>foo</span>
   <small>bar</small>
   <a href="#">qux</a>
</div>
```
```css
.container > * {
    display: inline-block;
    margin-right: 15px;
}
```

And then we were requested to add some microdata tags into this block.
E.g. http://schema.org/Person
```html
<div class="container">
   <span>foo</span>
   <small>bar</small>
   <meta itemprop="billingPeriod" content="2014-12-21/P30D" />
   <a href="#">qux</a>
</div>
```
Thus, we have double space between `<small>` and `<a>`

**Comment by rossPatton** [comment]
that's actually a very good use case @thybzi

**Comment by nayaabkhan** [comment]
@thybzi Yep a good case for targeting `meta`s. It should be quick to whitelist it.

**Comment by thybzi** [comment]
Similar cases could appear for `<script>` and `<style>`
